File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        
        ans=[]
        if root==None:
            return ans
        q=[]
        q.append(root)
        while q:
            level=[]
            n=len(q)
            for i in range(0,n,1):
                node=q.pop(0)
                # q.remove(q[0])
                level.append(node.val)
                if node.left !=None:
                    q.append(node.left)
                if node.right!=None:
                    q.append(node.right)
            ans.append(level)
        return ans

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root)
    {
        queue<TreeNode*> qu;
        vector<vector<int>> ans;
        if(root == nullptr)
        {
            return ans;
        }
        qu.push(root);
        while(!qu.empty())
        {
            int size = qu.size();
            vector<int> temp;
            for(int i = 0 ; i < size ; ++i)
            {
                temp.push_back(qu.front()->val);
                if(qu.front()->left)
                {
                    qu.push(qu.front()->left);
                }
                if(qu.front()->right)
                {
                    qu.push(qu.front()->right);
                }
                qu.pop();
            }
            ans.push_back(temp);
        }
        return ans;
    }
};